api/src/lib/helpers.ts

```diff
@@ -208,28 +378,29 @@ export const updateDOI = async (
                 ],
                 language: 'en',
                 types: {
-                    resourceTypeGeneral: 'Other',
+                    resourceTypeGeneral: publicationVersion.publication.type === 'PEER_REVIEW' ? 'PeerReview' : 'Other',
                     resourceType: publicationVersion.publication.type
                 },
-                relatedIdentifiers: allReferencesWithDOI,
+                relatedIdentifiers: [...linkedPublicationDOIs, ...referenceDOIs, publicationDOI],
                 relatedItems: otherReferences,
                 fundingReferences: publicationVersion.funders.map((funder) => ({
                     funderName: funder.name,
                     funderReference: funder.ror || funder.link,
                     funderIdentifierType: funder.ror ? 'ROR' : 'Other'
-                }))
+                })),
+                version: publicationVersion.versionNumber
             }
         }
     };
 
-    const doiRes = await axios.put<I.DOIResponse>(`${process.env.DATACITE_ENDPOINT as string}/${doi}`, payload, {
+    const doi = await axios.post<I.DOIResponse>(`${process.env.DATACITE_ENDPOINT}`, payload, {
         auth: {
             username: process.env.DATACITE_USER as string,
             password: process.env.DATACITE_PASSWORD as string
         }
     });
 
-    return doiRes.data;
+    return doi.data;
```

Comment:
The updatePublicationDOI and createPublicationVersionDOI functions are really similar. Can we just use one function and take a parameter to say whether we're dealing with a versionless / versioned DOI?

Reply:
I will need to update these helpers in [OC-391](https://jiscdev.atlassian.net/browse/OC-391) anyway with "isPreviousVersionOf" and "isNewVersionOf" and also updating the canonical DOI when creating new version. I was thinking to create a separate transformer/mapper helper function/s which can be used by both of these helpers when creating/updating DOIs. For this specific use case, If we merge all the logic inside a single helper function we will end up with a huge function that's doing lots of unrelated stuff inside based on the extra params. We should split them up as much as we can and keep each helper function as simple as possible.

Reply:
OK, sounds good also